service.go

```diff
@@ -79,10 +78,6 @@ type service struct {
 
 // NewService builds the Argus client service from the given configuration.
 func NewService(cfg Config, getLogger func(context.Context) *zap.Logger) (*service, error) {
-	if cfg.Logger == nil {
-		cfg.Logger = sallust.Default()
-	}
```

Comment:
these checks will be replaced with client config options in the near future (later this week)

Reply:
Instead of removing something useful, change what the cfg.Logger is set to.  Change to `zap.NewNop()` so the logger is never `nil`.  This makes the usage cleaner & removes about 1/2 of the needed changes in this PR.

Reply:
Definitely replace this with the functional options pattern, as this code is exactly what that pattern was designed to address.

It's also ok to let things panic in a library like this.  Remember, `ancla` is not a complete program. It requires an application around it to function. Just provide a functional option that sets the logger, and document that if one is not set that the service will panic.

The problem with every library setting a default logger (or any default dependency) is that every library will do it differently. Some might use `NewNop()`, but others might use `NewDevelopment()`.  Letting things panic results in less code and in less inconsistency.